packages/runtime/container-runtime/src/blobManager.ts

```diff
@@ -723,23 +741,24 @@ export class BlobManager extends TypedEventEmitter<IBlobManagerEvents> {
 			return [];
 		}
 
+		// This holds the storage ids for each deleted local id entry in the redirect table.
+		const maybeUnusedStorageIds: string[] = [];
 		// The routes or blob node paths are in the same format as returned in getGCData -
 		// `/<BlobManager.basePath>/<blobId>`.
 		for (const route of sweepReadyBlobRoutes) {
-			const pathParts = route.split("/");
-			assert(
-				pathParts.length === 3 && pathParts[1] === BlobManager.basePath,
-				0x586 /* Invalid blob node id in deleted routes. */,
-			);
-			const blobId = pathParts[2];
-			if (!this.redirectTable.has(blobId)) {
+			const blobId = getBlobIdFromGCNodePath(route);
+			const storageId = this.redirectTable.get(blobId);
+			if (storageId === undefined) {
```

Comment:
Redirect table could have `blobId` but with the value of `undefined`, if detached.  So in that case you changed the semantics of this check.

Reply:
Same goes for similar code in `updateUnusedRoutes`

Reply:
That's a good point and a good practice in general. However, note that the summarizer client will never be detached. It will also not have any local changes (some day) so this shouldn't be a problem.

Reply:
Also, the reason I went with get here is to avoid doing "undefined" check twice - once in `has` and then again on the return of `get`.